hbp_validation_framework/__init__.py

```diff
@@ -793,29 +788,45 @@ def edit_test(self, name="", test_id="", alias=None, version="", author="",
                                       test_type="network structure", score_type="Other", protocol="To be filled sometime later", data_location="collab://Validation Framework/observations/test_data/cell_density_Halasy_1996.json", data_type="Mean, SD")
         """
 
-        values = self.get_attribute_options()
-
-        if species not in values["species"]:
-            raise Exception("species = '" +species+"' is invalid.\nValue has to be one of these: " + str(values["species"]))
-        if brain_region not in values["brain_region"]:
-            raise Exception("brain_region = '" +brain_region+"' is invalid.\nValue has to be one of these: " + str(values["brain_region"]))
-        if cell_type not in values["cell_type"]:
-            raise Exception("cell_type = '" +cell_type+"' is invalid.\nValue has to be one of these: " + str(values["cell_type"]))
-        if data_modality not in values["data_modalities"]:
-            raise Exception("data_modality = '" +data_modality+"' is invalid.\nValue has to be one of these: " + str(values["data_modality"]))
-        if test_type not in values["test_type"]:
-            raise Exception("test_type = '" +test_type+"' is invalid.\nValue has to be one of these: " + str(values["test_type"]))
-        if score_type not in values["score_type"]:
-            raise Exception("score_type = '" +score_type+"' is invalid.\nValue has to be one of these: " + str(values["score_type"]))
-
-        if alias == "":
-            alias = None
+        if test_id == "":
+            raise Exception("Test ID needs to be provided for editing a test.")
 
         id = test_id   # as needed by API
         test_data = locals()
-        for key in ["self", "test_id", "values"]:
+        for key in ["self", "test_id"]:
             test_data.pop(key)
 
+        # assign existing values for parameters not specified
```

Comment:
This shouldn't really be necessary. The PUT url should accept missing parameters. Or we could define a PATCH endpoint.

Reply:
Please could you open a ticket in the VF repository for modifying the API accordingly?

Reply:
Done.... https://github.com/HumanBrainProject/hbp-validation-framework/issues/181